client/db/bolt/upgrades.go

```diff
@@ -85,8 +88,17 @@ func (db *BoltDB) upgradeDB() error {
 
 	return db.Update(func(tx *bbolt.Tx) error {
 		// Execute all necessary upgrades in order.
-		for _, upgrade := range upgrades[version:] {
-			err := upgrade(tx)
+		for i, upgrade := range upgrades[version:] {
+			// err := upgrade(tx)
+			// if err != nil {
+			// 	return fmt.Errorf("error upgrading DB: %v", err)
+			// }
+			// // Persist the database version.
+			// err = setDBVersion(tx, uint32(i+1))
+			// if err != nil {
+			// 	return fmt.Errorf("error setting DB version: %v", err)
+			// }
+			err := doUpgrade(tx, upgrade, uint32(i+1))
```

Comment:
Thinking about the value of `i` in this loop for different values of `version`... unless I'm mistaken, `uint32(i+1)` here is appropriate only for when the starting `version` is 0.

Say we have versions all the way up to 10, and we start up with v5 db file, then in `for i, upgrade := range upgrades[version:]` the first iteration of the loop will be `i=0`, telling `doUpgrade` that `newVersion` is 1, but it needs to be 6.
Even for a v1 db, it seems it will set the version to 1 after the v2 upgrade.  Perhaps a quick test of the `upgradeDB` function?

Reply:
Obvious now that you point it out. When I look again, I see `[INF] CORE[DB]: Upgrading database from version 1 to 2` every time I start the app, meaning that after the upgrade the version is being set to `1` again.